UT-DemoTests/RepoTests.swift

```diff
@@ -0,0 +1,61 @@
+//
+//  RepoTests.swift
+//  UT-DemoTests
+//
+//  Created by maithigiang on 8/9/19.
+//  Copyright © 2019 maithigiang. All rights reserved.
+//
+
+import XCTest
+@testable import UT_Demo
+
+class RepoTests: XCTestCase {
+
+    override func setUp() {
+        // Put setup code here. This method is called before the invocation of each test method in the class.
+    }
+
+    override func tearDown() {
+        // Put teardown code here. This method is called after the invocation of each test method in the class.
+    }
+
+    func testRepo_ValidJson_CanBeInstantiated() {
+        guard let url = Bundle(for: type(of: self)).url(forResource: "validJson", withExtension: "json"),
+            let data = try? Data(contentsOf: url, options: .mappedIfSafe) else {
+            XCTFail("Missing file: validJson.json")
+            return
+        }
+        let repo = Repo(jsonObject: data)
+        XCTAssertNotNil(repo?.id)
+    }
+    
+    func testParserJson() {
+        guard let url = Bundle(for: type(of: self)).url(forResource: "validJson", withExtension: "json"),
+            let data = try? Data(contentsOf: url, options: .mappedIfSafe) else {
+            XCTFail("Missing file: validJson.json")
+            return
+        }
+        let repo = Repo(jsonObject: data)
+        XCTAssertEqual(repo!.fullname, "vsouza/awesome-ios")
+        XCTAssertEqual(repo!.id, 21700699)
+    }
```

Comment:
// arrange
            if let url = Bundle(for: type(of: self)).url(forResource: "validJson", withExtension: "json"),
            let data = try? Data(contentsOf: url, options: .mappedIfSafe) {
                    // act
                    let repo = Repo(jsonObject: data)
                   
                    // assert
                    XCTAssertEqual(repo?.fullname, "vsouza/awesome-ios")
                    XCTAssertEqual(repo?.id, 21700699)
           } else {
                  XCTFail("Missing file: validJson.json")
           }
Cái này ko cần test đoạn json đầu vào đâu vì nó là data test của mình mà người test tự kiểm soát. Viết như em cũng ko sai, chỉ cần bỏ cái XCTFail("Missing file: validJson.json"), thêm struct arrange - act - assert và chú ý repo?.fullname nhá